juju/juju
# Status time

## Description of change

The following change adds the status-time to all outputs that are
in the long form (json, yaml, tabular), ignoring short form for now.

Most of the work includes changing the tests to acknowledge the new
field. Including the more generic approach of substituting the fake
time stamp from the since one, to the now parameterised key.

One thing I'm not 100% settled on and open for questions is the
following:

 1. Naming of the field
 1. Should the field be nested inside a object. Think `meta.status-time`
 1. Should we gain the status from the API instead of using time.Now()

### Other:

Inconsistencies around long form outputs especially between json, yaml
and tabular, where by the later doesn't show the since time in the output.
Should all the outputs show this?

> Why is this change needed?

The change is added to allow more context when sharing or diagnosing 
problems between setups i.e. understanding when something happened.

## QA steps

> How do we verify that the change works?

Running `juju status` should reveal the new time-since status.

## Documentation changes

> Does it affect current user workflow? CLI? API?

It does change the output to the CLI and we should think about if this
does affect anyone who consumes the data.

## Bug reference

> Does this change fix a bug? Please add a link to it.

https://bugs.launchpad.net/juju/+bug/1765404

## Demo

[![demo](https://asciinema.org/a/178473.png)](https://asciinema.org/a/178473?autoplay=1)
